elephant/causality/granger.py

```diff
@@ -596,3 +781,173 @@ def conditional_granger(signals, max_order, information_criterion='aic'):
+        If None, it will be determined from other parameters.
+        Default: None
+    overlap : float, optional
+        Overlap between segments represented as a float number between 0 (no
+        overlap) and 1 (complete overlap).
+        Default: 0.5 (half-overlapped)
+    num_iterations : int
+        Number of iterations for algorithm to estimate spectral factorization.
+        Default: 300
+    term_crit : float
+        Termination criterion for iteration step in spectral matrix
+        factorization
+        Default: 1e-12
+
+    Returns
+    -------
+    freqs : np.ndarray
+        Frequencies associated with the spectral Granger Causality estimate.
+    Causality
+        A `namedtuple` with the following attributes:
+            directional_causality_x_y : np.ndarray
+                Spectrally resolved Granger causality influence of `signal_i`,
+                abbreviated by  X, onto `signal_j`, abbreviated by Y.
+
+            directional_causality_y_x : np.ndarray
+                Spectrally resolved Granger causality influence of `signal_j`,
+                abbreviated by  Y, onto `signal_i`, abbreviated by Y.
+
+            instantaneous_causality : np.ndarray
+                The remaining spectrally resolved channel interdependence not
+                accounted for by the directional causalities (e.g. shared input
+                to X, i.e. `signal_i`, and Y, i.e. `signal_j`).
+
+            total_interdependence : np.ndarray
+                The sum of the former three metrics. It measures the dependence
+                of X, i.e. `signal_i` and Y, i.e. `signal_j`. If the total
+                interdependence is positive, X and Y are not independent.
+    """
+    if isinstance(signal_i, neo.core.AnalogSignal) and \
+            isinstance(signal_j, neo.core.AnalogSignal):
+        signals = signal_i.merge(signal_j)
+    elif isinstance(signal_i, np.ndarray) and isinstance(signal_j, np.ndarray):
+        signals = np.vstack([signal_i, signal_j])
+
+    # Calculate cross spectrum for signals
+    freqs, S = segmented_multitaper_cross_spectrum(
+        signals=signals, n_segments=n_segments, len_segment=len_segment,
+        frequency_resolution=frequency_resolution, overlap=overlap, fs=fs,
+        nw=nw, num_tapers=num_tapers, peak_resolution=peak_resolution,
+        return_onesided=False)
+
+    # Remove units attached by the multitaper_cross_spectrum
+    if isinstance(S, pq.Quantity):
+        S = S.magnitude
+
+    # Transpose cross spectrum due to different conventions used in
+    # segemented_multitaper_cross_spectrum and the calculations of spectral
+    # Granger causality
+    S = np.transpose(S, axes=(1, 0, 2))
+
+    # Move frequencies to first axis - Needed for _spectral_factorization
+    S = np.transpose(S, axes=(2, 0, 1))
+
+    # Decompose cross spectrum into covariance and transfer function
+    C, H = _spectral_factorization(S, num_iterations=num_iterations)
```

Comment:
[Optional, Question] 
Why is the parameter `term_crit` of `pairwise_spectral_granger` not passed as an argument to `_spectral_factorization` ?, like:
```python
    C, H = _spectral_factorization(S, num_iterations=num_iterations, term_crit=term_crit)
```

If I get this correctly `_spectral_factorization` will always use the default value `1e-12` , is this intended? (independent of `term_crit` passed to `pairwise_spectral_granger`)